src/Uno.Extensions.Validation/Validator.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace Uno.Extensions.Validation;
+
+internal class Validator : IValidator
+{
+	private readonly IServiceProvider _services;
+	private IInstanceType? TypedValidator(Type instanceType) => _services.GetServices<IInstanceType>().FirstOrDefault(x => x.InstanceType == instanceType);
```

Comment:
I think the return value here can just be IValidator? since this is what we're interested in